pandas/core/generic.py

```diff
@@ -4918,9 +4920,7 @@ def reindex(self: NDFrameT, *args, **kwargs) -> NDFrameT:
             for axis, ax in axes.items()
             if ax is not None
         ):
-            if copy:
-                return self.copy()
-            return self
+            return self.copy(deep=copy)
```

Comment:
with copy=False we no longer get a new object.  i generally like this, but its an API change

Reply:
Ah, good point. I shouldn't change this in this PR (or at least not for BlockManager-based dataframe). Now for the CoW I need to pass down the copy=None, so that will unfortunately require a check for AM vs BM then.